src/model.ts

```diff
@@ -518,6 +523,35 @@ export function build<TSchema extends BaseSchema, TDefaults extends Partial<TSch
     }
   );
 
+  /**
+   * @description
+   * Performs an optimized `find` to test for the existence of any document matching the filter criteria.
+   *
+   * @param filter {Filter<TSchema>}
+   * @param [options] {FindOptions<TSchema>}
+   *
+   * @returns {Promise<boolean>}
+   */
+  model.exists = async (filter, options) => {
```

Comment:
Please write this function as a named function. And also add the types to the arguments here, even though they're infered by TS.